### PR TITLE
 Python implementation for the Rabin Karp algo. issue#2019

### LINKED ISSUE #2019: Python Implementation of Rabin Karp Algorithm
@jainaman224 
There is no Python implementation for the Rabin Karp algorithm. I would like to work on this issue for GSSoC. Please assign me with this issue.

### PR DESCRIPTION
**Pull request by Amitsharma45**
Fixes #2019 #### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [ ] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- 
- 

#### Languages Used:
- PYTHON
- 

#### Files Added:
- 
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by raghav-dalmia** [comment]
save the file with .py extension. squash commits.

File: Python implementation for the Rabin Karp algorithm.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+			j+= 1
+			
+			if j == M: 
+				print ("Pattern found at index " + str(i)) 
+
+		 
+		if i < N-M: 
+			t = (d*(t-ord(txt[i])*h) + ord(txt[i + M]))% q 
+
```
Comment (raghav-dalmia) [inline Python implementation for the Rabin Karp algorithm.py]: remove these extra lines

File: Python implementation for the Rabin Karp algorithm.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+d = 256
+def search(pat, txt, q): 
+	M = len(pat) 
+	N = len(txt) 
+	i = 0
+	j = 0
+	p = 0 
+	t = 0  
+	h = 1
+
```
Comment (raghav-dalmia) [inline Python implementation for the Rabin Karp algorithm.py]: here too

File: Python implementation for the Rabin Karp algorithm.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+d = 256
+def search(pat, txt, q): 
+	M = len(pat)
```
Comment (raghav-dalmia) [inline Python implementation for the Rabin Karp algorithm.py]: use proper variable names

File: Python implementation for the Rabin Karp algorithm.py (hunk 1 above)
Comment (Amitsharma45) [reply Python implementation for the Rabin Karp algorithm.py]: ok

File: Python implementation for the Rabin Karp algorithm.py (hunk 3 above)
Comment (Amitsharma45) [reply Python implementation for the Rabin Karp algorithm.py]: done ,please check.

File: Python implementation for the Rabin Karp algorithm.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
```
Comment (SakshiShreya) [inline Python implementation for the Rabin Karp algorithm.py]: Use 4 space indentation.

File: Python implementation for the Rabin Karp algorithm.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
+d = 256
+def search(pat, txt, q): 
+	M = len(pat) 
+	N = len(txt) 
+	i = 0
+	j = 0
+	p = 0 
+	t = 0  
+	h = 1
+	for i in range(M-1):
```
Comment (SakshiShreya) [inline Python implementation for the Rabin Karp algorithm.py]: Add space around operators.

File: Python implementation for the Rabin Karp algorithm.py (hunk 4 above)
Comment (Amitsharma45) [reply Python implementation for the Rabin Karp algorithm.py]: done

**Comment by somya-kapoor** [comment]
@Amitsharma45 Are you working on it?

**Comment by Amitsharma45** [comment]
yes

**Comment by Amitsharma45** [comment]
yes @somya-kapoor

**Comment by Amitsharma45** [comment]
@raghav-dalmia @somya-kapoor @SakshiShreya  Please check my pr

File: Python implementation for the Rabin Karp algorithm.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
+d = 256
```
Comment (somya-kapoor) [inline Python implementation for the Rabin Karp algorithm.py:1]: Add comments

File: Python implementation for the Rabin Karp algorithm.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+def search(pat, txt, q): 
+    M = len(pat) 
+    N = len(txt) 
+    i = 0
+    j = 0
+    p = 0 
+    t = 0  
+    h = 1
+    for i in range(M-1):
```
Comment (somya-kapoor) [inline Python implementation for the Rabin Karp algorithm.py:10]: Add space around the operator

File: Python implementation for the Rabin Karp algorithm.py [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+def search(pat, txt, q): 
+    M = len(pat) 
+    N = len(txt) 
+    i = 0
+    j = 0
+    p = 0 
+    t = 0  
+    h = 1
+    for i in range(M-1): 
+        h = (h * d)% q
```
Comment (somya-kapoor) [inline Python implementation for the Rabin Karp algorithm.py:11]: Add space around the operator

File: Python implementation for the Rabin Karp algorithm.py [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    i = 0
+    j = 0
+    p = 0 
+    t = 0  
+    h = 1
+    for i in range(M-1): 
+        h = (h * d)% q  
+    for i in range(M): 
+        p = (d * p + ord(pat[i]))% q 
+	t = (d * t + ord(txt[i]))% q
```
Comment (somya-kapoor) [inline Python implementation for the Rabin Karp algorithm.py:14]: inconsistency in indentation

File: Python implementation for the Rabin Karp algorithm.py [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+        if i < N-M: 
+	    t = (d*(t-ord(txt[i])*h) + ord(txt[i + M]))% q 
+	    if t < 0: 
+	        t = t + q 
+txt = input("Enter text\n")
+pat = input("Enter pattern\n")
+q = 101 
+search(pat, txt, q) 
+# This code is contributed by Amit sharma 
+'''
```
Comment (somya-kapoor) [inline Python implementation for the Rabin Karp algorithm.py:32]: leave a line before input

**Comment by somya-kapoor** [comment]
@Amitsharma45 Please do the requested changes

**Comment by codermuskanj** [comment]
@somya-kapoor @raghav-dalmia @SakshiShreya @jainaman224 is plagiarism allowed? The whole code (even the variables) of @Amitsharma45 has been copied from geeks for geeks only example is different.

**Pull request closed without merging**